File: app/src/main/java/com/onedelay/mymovie/fragment/PosterFragment.java
package com.onedelay.mymovie.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.onedelay.mymovie.Constants;
import com.onedelay.mymovie.R;
import com.onedelay.mymovie.database.MovieEntity;

public class PosterFragment extends Fragment {
    PosterFragmentCallback callback;

    public interface PosterFragmentCallback {
        void onChangeFragment();

        void setData(int id, String title, int grade, float rating);
    }

    public static PosterFragment newInstance(int index, MovieEntity item) {
        PosterFragment fragment = new PosterFragment();
        Bundle bundle = new Bundle();
        bundle.putInt(Constants.KEY_INDEX, index);
        bundle.putInt(Constants.KEY_MOVIE_ID, item.getId());
        bundle.putString(Constants.KEY_IMAGE_URL, item.getImage());
        bundle.putString(Constants.KEY_TITLE, item.getTitle());
        bundle.putFloat(Constants.KEY_RATE, item.getReservation_rate()); // 예매율
        bundle.putInt(Constants.KEY_GRADE, item.getGrade());
        bundle.putString(Constants.KEY_DATE, item.getDate());
        bundle.putFloat(Constants.KEY_RATING, item.getAudience_rating());
        fragment.setArguments(bundle);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.fragment_poster, container, false);

        ImageView imageView = rootView.findViewById(R.id.list_image);
        TextView title = rootView.findViewById(R.id.list_title);
        TextView rate = rootView.findViewById(R.id.rate);
        TextView grade = rootView.findViewById(R.id.grade);

        if (getArguments() != null) {
            Bundle bundle = getArguments();
            Glide.with(this).load(bundle.getString(Constants.KEY_IMAGE_URL)).into(imageView);
            title.setText(String.format(getString(R.string.list_fragment_title), bundle.getInt(Constants.KEY_INDEX), bundle.getString(Constants.KEY_TITLE)));
            rate.setText(String.format(getString(R.string.list_fragment_rate), bundle.getFloat(Constants.KEY_RATE)));
            grade.setText(String.format(getString(R.string.list_fragment_grade), bundle.getInt(Constants.KEY_GRADE)));
        }

        rootView.findViewById(R.id.detailButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // 상세보기 버튼을 클릭하면 viewPager 를 숨기고 DetailFragment 를 띄우도록 콜백 메소드를 호출한다.
                callback.onChangeFragment();
                if (getArguments() != null) {
                    Bundle bundle = getArguments();
                    callback.setData(bundle.getInt(Constants.KEY_MOVIE_ID), bundle.getString(Constants.KEY_TITLE), bundle.getInt(Constants.KEY_GRADE), bundle.getFloat(Constants.KEY_RATING));
                }
            }
        });

        return rootView;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        if (context instanceof PosterFragmentCallback) {
            callback = (PosterFragmentCallback) context;
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();

        if (callback != null) callback = null;
    }
}

File: app/src/main/java/com/onedelay/mymovie/fragments/PosterFragment.kt
package com.onedelay.mymovie.fragments

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.onedelay.mymovie.Constants
import com.onedelay.mymovie.R
import com.onedelay.mymovie.data.local.entity.MovieEntity
import kotlinx.android.synthetic.main.fragment_poster.view.*

class PosterFragment : Fragment() {
    private var callback: PosterFragmentCallback? = null

    interface PosterFragmentCallback {
        fun onChangeFragment(bundle: Bundle)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_poster, container, false) as ViewGroup

        arguments?.run {
            Glide.with(this@PosterFragment).load(getString(Constants.KEY_IMAGE_URL)).into(rootView.list_image)
            rootView.list_title.text = String.format(getString(R.string.list_fragment_title), getInt(Constants.KEY_INDEX), getString(Constants.KEY_TITLE))
            rootView.rate.text = String.format(getString(R.string.list_fragment_rate), getFloat(Constants.KEY_RATE))
            rootView.grade.text = String.format(getString(R.string.list_fragment_grade), getInt(Constants.KEY_GRADE))
        }

        rootView.detailButton.setOnClickListener {
            // 상세보기 버튼을 클릭하면 viewPager 를 숨기고 DetailFragment 를 띄우도록 콜백 메소드를 호출한다.
            arguments?.let { callback?.onChangeFragment(it) }
        }

        return rootView
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        if (context is PosterFragmentCallback) {
            callback = context
        }
    }

    override fun onDetach() {
        super.onDetach()

        if (callback != null) callback = null
    }

    companion object {
        fun newInstance(index: Int, item: MovieEntity): PosterFragment {
            return PosterFragment().apply {
                arguments = Bundle().apply {
                    putInt(Constants.KEY_INDEX, index)
                    putInt(Constants.KEY_MOVIE_ID, item.id)
                    putString(Constants.KEY_IMAGE_URL, item.image)
                    putString(Constants.KEY_TITLE, item.title)
                    putFloat(Constants.KEY_RATE, item.reservation_rate) // 예매율
                    putInt(Constants.KEY_GRADE, item.grade)
                    putString(Constants.KEY_DATE, item.date)
                    putFloat(Constants.KEY_RATING, item.audience_rating)
                }
            }
        }
    }
}
